mantidproject/mantid
# Enabled progress reporting in Engineering Diffraction GUI

Previously, when running calibration and focus in the ED GUI, the system appeared to hang for a very long time without really doing anything. This is because the algorithms used (`EnggCalibrate`, `EnggVanadiumCorrections`, `EnggFitPeaks` and their children) weren't reporting to the `Progress` object properly. They have now been wired up correctly.

Extensive refactoring was also done in the associated Python files to bring them in line with Mantid Python standards (sorry for the huge diffs).

**To test:**
[Re-enable the ED GUI](https://github.com/mantidproject/mantid/commit/7304420b4326d6cea1a19523fe3d90b3691151a0)

Go to Interfaces->Diffraction->Engineering Diffraction.
From the Calibration tab, in the 'Make new calibration' area, load `ENGINX00275775.nxs` for the vanadium and `ENGINX00277171.nxs` for the calibration (or similar data of your choosing), then hit 'Calibrate', Make sure that the progress bar updates all the way through execution (you'll know it's done when the bar at bottom of the GUI displays a 'ready' message).

From the Focus tab, in the 'Focus run' area, load a run from the same cycle (such as `ENGINX00277208.nxs`), select which banks to focus (up to you) and hit 'Focus'. Again, the progress bar should update all the way through.

Focussing should have given you some data that you can then fit in the Fitting tab. Do a fit and, again, make sure progress is reported properly.

Fixes #21024 

**Release Notes**
[Here](https://github.com/mantidproject/mantid/commit/21c7d19e4b569395b0259e68d0b652965e7d62c9) 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
